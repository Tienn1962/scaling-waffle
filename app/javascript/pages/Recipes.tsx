import React from "react";

export default () => (
  <main>
    <h1>Here's a bunch of recipes :</h1>
  </main>
);
