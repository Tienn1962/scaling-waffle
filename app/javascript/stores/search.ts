import { PayloadAction, configureStore, createSlice } from "@reduxjs/toolkit";

export interface SearchState {
  filters: string[];
  page: number;
}

const initialState = { filters: [], page: 0 } as SearchState;

const searchSlice = createSlice({
  name: "search",
  initialState,
  reducers: {
    addFilter(state, action: PayloadAction<string>) {
      if (state.filters.includes(action.payload)) {
        return;
      }

      state.filters = [...state.filters, action.payload];
      state.page = 0;
    },

    removeFilter(state, action: PayloadAction<string>) {
      state.filters = [...state.filters.filter((f) => f !== action.payload)];
      state.page = 0;
    },

    setPage(state, action: PayloadAction<number>) {
      state.page = action.payload;
    },
  },
});

export const { addFilter, removeFilter, setPage } = searchSlice.actions;
export default searchSlice;
export const store = configureStore({
  reducer: searchSlice.reducer,
});
