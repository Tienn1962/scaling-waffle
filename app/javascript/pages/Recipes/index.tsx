import React, { useEffect, useState } from "react";
import { PlusCircleIcon } from "@heroicons/react/outline";
import RecipeCard from "../../components/RecipeCard";
import PaginatedList from "../../components/PaginatedList";
import Loading from "../../components/Loading";
import IngredientBadges from "../../components/IngredientBadges";
import axios from "axios";
import { Link } from "react-router-dom";
import Show from "../../components/Show";
import search, { SearchState } from "../../stores/search";
import { useDispatch, useSelector } from "react-redux";

export default () => {
  const [loading, setLoading] = useState(false);
  const [recipes, setRecipes] = useState<any[]>([]);
  const [count, setCount] = useState<number>(0);
  const [limit, setLimit] = useState<number>(0);
  const [searchValue, setSearchValue] = useState<string>("");
  const filters = useSelector((state: SearchState) => state.filters);
  const page = useSelector((state: SearchState) => state.page);
  const dispatch = useDispatch();

  const fetchRecipes = async () => {
    setLoading(true);

    const response = await axios.get("/api/v1/recipes", {
      params: {
        page: page,
        ingredients: filters,
      },
    });

    setRecipes(response.data.recipes);
    setCount(response.data.pagination.count);
    setLimit(response.data.pagination.limit);
    setLoading(false);
  };

  const onPageChange = (page: number) => {
    dispatch(search.actions.setPage(page));
  };

  const addFilter = (e) => {
    e.preventDefault();

    if (!!searchValue) {
      dispatch(search.actions.addFilter(searchValue));
    }
    setSearchValue("");
  };

  const removeFilter = (filter: string) => {
    dispatch(search.actions.removeFilter(filter));
  };

  useEffect(() => {
    fetchRecipes();
  }, [filters, page]);

  return (
    <main className="m-60 flex flex-col items-center justify-center">
      <h1 className="mb-5 text-4xl">Recipes</h1>
      <form
        className="join"
        onSubmit={addFilter}
      >
        <input
          type="text"
          className="input join-item bg-black"
          placeholder="Search a recipe by ingredient"
          value={searchValue}
          onChange={(e) => {
            // prevent user from entering numbers
            setSearchValue(e.target.value.replace(/[^a-z]/gi, ""));
          }}
        />
        <button
          className="btn btn-success no-animation join-item"
          type="submit"
        >
          <PlusCircleIcon className="w-6 h-6" />
        </button>
      </form>

      <div className="m-10">
        <IngredientBadges
          badges={filters}
          onClick={removeFilter}
        />
      </div>

      <PaginatedList
        data={recipes}
        count={count}
        limit={limit}
        page={page}
        onPageChange={onPageChange}
        item={(data) => (
          <Link
            key={data.id}
            to={`/recipes/${data.id}`}
            className="m-5"
          >
            <RecipeCard
              title={data.title}
              image_url={data.image_url}
              author={data.author}
              ratings={data.ratings}
              prepTime={data.prep_time}
              cookTime={data.cook_time}
              ingredients={data.ingredients}
            ></RecipeCard>
          </Link>
        )}
      ></PaginatedList>

      <Show if={loading}>
        <Loading />
      </Show>
    </main>
  );
};
