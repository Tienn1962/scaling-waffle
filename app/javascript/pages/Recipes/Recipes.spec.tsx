import { act, render } from "@testing-library/react";
import axios from "axios";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import Recipes from ".";
import { Provider } from "react-redux";
import { store } from "../../stores/search";

jest.mock("axios");

const recipes = [
  {
    id: 1,
    author: "John Recipe",
    prep_time: 10,
    cook_time: 10,
    ratings: 5.0,
    title: "Potatoes with chicken",
    ingredients: [{ title: "3 potatoes" }, { title: "chicken breast" }],
  },
  {
    id: 2,
    author: "Jane Recipe",
    prep_time: 10,
    cook_time: 10,
    ratings: 5.0,
    title: "Tomatoes with beef",
    ingredients: [{ title: "3 tomatoes" }, { title: "beef" }],
  },
];

const response = { data: { recipes, pagination: { count: 2, limit: 1 } } };

(axios.get as jest.Mock).mockResolvedValue(response);

it("renders a list of recipes", async () => {
  const component = await act(async () =>
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Recipes />
        </MemoryRouter>
      </Provider>
    )
  );

  expect(component).toMatchSnapshot();
});
