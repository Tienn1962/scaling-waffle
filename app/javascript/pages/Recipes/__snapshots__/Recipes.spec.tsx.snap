// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of recipes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="m-60 flex flex-col items-center justify-center"
      >
        <h1
          class="mb-5 text-4xl"
        >
          Recipes
        </h1>
        <form
          class="join"
        >
          <input
            class="input join-item bg-black"
            placeholder="Search a recipe by ingredient"
            type="text"
            value=""
          />
          <button
            class="btn btn-success no-animation join-item"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 9v3m0 0v3m0-3h3m-3 0H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </form>
        <div
          class="m-10"
        >
          <div
            class="flex flex-row items-center"
          >
            <span
              class="mr-3"
            >
              Ingredients :
            </span>
            <div
              class="badge bg-black"
            >
              any
            </div>
          </div>
        </div>
        <div
          class="flex flex-row flex-wrap justify-center"
        >
          <a
            class="m-5"
            href="/recipes/undefined"
          >
            <div
              class="w-60 card bg-neutral text-neutral-content"
            >
              <figure>
                <img
                  alt="Potatoes with chicken"
                  class="object-fit h-60"
                  src="assets/No-Image-Placeholder.png"
                />
              </figure>
              <div
                class="card-body"
              >
                <h2
                  class="card-title"
                >
                  Potatoes with chicken
                </h2>
                <div
                  class="flex"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <span
                    class="pl-2"
                  >
                    John Recipe
                  </span>
                </div>
                <div
                  class="flex"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <span
                    class="pl-2"
                  >
                    10
                    m | 
                    10
                    m
                  </span>
                </div>
                <div
                  class="flex"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <span
                    class="pl-2"
                  >
                    5
                    /5
                  </span>
                </div>
                <div
                  class="flex"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <span
                    class="pl-2"
                  >
                    2
                     ingredients
                  </span>
                </div>
              </div>
            </div>
          </a>
          <a
            class="m-5"
            href="/recipes/undefined"
          >
            <div
              class="w-60 card bg-neutral text-neutral-content"
            >
              <figure>
                <img
                  alt="Tomatoes with beef"
                  class="object-fit h-60"
                  src="assets/No-Image-Placeholder.png"
                />
              </figure>
              <div
                class="card-body"
              >
                <h2
                  class="card-title"
                >
                  Tomatoes with beef
                </h2>
                <div
                  class="flex"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <span
                    class="pl-2"
                  >
                    Jane Recipe
                  </span>
                </div>
                <div
                  class="flex"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <span
                    class="pl-2"
                  >
                    10
                    m | 
                    10
                    m
                  </span>
                </div>
                <div
                  class="flex"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <span
                    class="pl-2"
                  >
                    5
                    /5
                  </span>
                </div>
                <div
                  class="flex"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <span
                    class="pl-2"
                  >
                    2
                     ingredients
                  </span>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="flex flex-row justify-center"
        >
          <ul
            aria-label="Pagination"
            class="join fixed bottom-0"
            role="navigation"
          >
            <li
              class="join-item btn btn-square no-animation disabled"
            >
              <a
                aria-disabled="true"
                aria-label="Previous page"
                class=" "
                rel="prev"
                role="button"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 19l-7-7m0 0l7-7m-7 7h18"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </a>
            </li>
            <li
              class="selected"
            >
              <a
                aria-current="page"
                aria-label="Page 1 is your current page"
                class="join-item btn btn-square no-animation bg-black"
                rel="canonical"
                role="button"
                tabindex="-1"
              >
                1
              </a>
            </li>
            <li>
              <a
                aria-label="Page 2"
                class="join-item btn btn-square no-animation"
                rel="next"
                role="button"
                tabindex="0"
              >
                2
              </a>
            </li>
            <li
              class="join-item btn btn-square no-animation"
            >
              <a
                aria-disabled="false"
                aria-label="Next page"
                class=""
                rel="next"
                role="button"
                tabindex="0"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14 5l7 7m0 0l-7 7m7-7H3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="m-60 flex flex-col items-center justify-center"
    >
      <h1
        class="mb-5 text-4xl"
      >
        Recipes
      </h1>
      <form
        class="join"
      >
        <input
          class="input join-item bg-black"
          placeholder="Search a recipe by ingredient"
          type="text"
          value=""
        />
        <button
          class="btn btn-success no-animation join-item"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 9v3m0 0v3m0-3h3m-3 0H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </form>
      <div
        class="m-10"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="mr-3"
          >
            Ingredients :
          </span>
          <div
            class="badge bg-black"
          >
            any
          </div>
        </div>
      </div>
      <div
        class="flex flex-row flex-wrap justify-center"
      >
        <a
          class="m-5"
          href="/recipes/undefined"
        >
          <div
            class="w-60 card bg-neutral text-neutral-content"
          >
            <figure>
              <img
                alt="Potatoes with chicken"
                class="object-fit h-60"
                src="assets/No-Image-Placeholder.png"
              />
            </figure>
            <div
              class="card-body"
            >
              <h2
                class="card-title"
              >
                Potatoes with chicken
              </h2>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  John Recipe
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  10
                  m | 
                  10
                  m
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  5
                  /5
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  2
                   ingredients
                </span>
              </div>
            </div>
          </div>
        </a>
        <a
          class="m-5"
          href="/recipes/undefined"
        >
          <div
            class="w-60 card bg-neutral text-neutral-content"
          >
            <figure>
              <img
                alt="Tomatoes with beef"
                class="object-fit h-60"
                src="assets/No-Image-Placeholder.png"
              />
            </figure>
            <div
              class="card-body"
            >
              <h2
                class="card-title"
              >
                Tomatoes with beef
              </h2>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  Jane Recipe
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  10
                  m | 
                  10
                  m
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.049 2.927c.3-.921 1.603-.921 1.902 0l1.519 4.674a1 1 0 00.95.69h4.915c.969 0 1.371 1.24.588 1.81l-3.976 2.888a1 1 0 00-.363 1.118l1.518 4.674c.3.922-.755 1.688-1.538 1.118l-3.976-2.888a1 1 0 00-1.176 0l-3.976 2.888c-.783.57-1.838-.197-1.538-1.118l1.518-4.674a1 1 0 00-.363-1.118l-3.976-2.888c-.784-.57-.38-1.81.588-1.81h4.914a1 1 0 00.951-.69l1.519-4.674z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  5
                  /5
                </span>
              </div>
              <div
                class="flex"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <span
                  class="pl-2"
                >
                  2
                   ingredients
                </span>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="flex flex-row justify-center"
      >
        <ul
          aria-label="Pagination"
          class="join fixed bottom-0"
          role="navigation"
        >
          <li
            class="join-item btn btn-square no-animation disabled"
          >
            <a
              aria-disabled="true"
              aria-label="Previous page"
              class=" "
              rel="prev"
              role="button"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 19l-7-7m0 0l7-7m-7 7h18"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li
            class="selected"
          >
            <a
              aria-current="page"
              aria-label="Page 1 is your current page"
              class="join-item btn btn-square no-animation bg-black"
              rel="canonical"
              role="button"
              tabindex="-1"
            >
              1
            </a>
          </li>
          <li>
            <a
              aria-label="Page 2"
              class="join-item btn btn-square no-animation"
              rel="next"
              role="button"
              tabindex="0"
            >
              2
            </a>
          </li>
          <li
            class="join-item btn btn-square no-animation"
          >
            <a
              aria-disabled="false"
              aria-label="Next page"
              class=""
              rel="next"
              role="button"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 5l7 7m0 0l-7 7m7-7H3"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
